File: py3/101.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def isSymmetric(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """

        def is_mirror(t1, t2):
            if t1 is None and t2 is None:
                return True
            elif t1 is None or t2 is None:
                return False
            elif t1.val == t2.val:
                left_right = is_mirror(t1.left, t2.right)
                right_left = is_mirror(t1.right, t2.left)
                if left_right and right_left:
                    # print("TRUE")
                    return True
                    
                else: 
                    return False
        
        return is_mirror(root, root)

File: go/src/101.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
	if root == nil {
		return true
	}

	return recursion(root.Left, root.Right)
}

func recursion(left *TreeNode, right *TreeNode) bool {
	if left == nil && right == nil {
		return true
	} else if left == nil || right == nil {
		return false
	}

	if left.Val != right.Val {
		return false
	}

	return recursion(left.Left, right.Right) &&
		recursion(left.Right, right.Left)
}